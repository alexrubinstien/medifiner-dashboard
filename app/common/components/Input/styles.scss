@import './app/common/styles/utils/_vars.scss';

.input {
  width: 195px;

  &--wide {
    width: 100%;
  }

  &__label {
    position: relative;
    top: -4px;
    display: inline-block;
    color: $text-color;
    font-size: 12px;
    font-weight: 700;
    line-height: 20px;
    letter-spacing: 1.5px;
    text-transform: uppercase;
  }

  &__input {
    width: 100%;
    padding: 7px 0;
    background: transparent;
    border: 0;
    border-bottom: 2px solid $border-color;
    color: $primary-color;
    font-size: 16px;
    font-weight: 500;
    line-height: 24px;
    letter-spacing: 1px;

    &:-moz-placeholder {
      color: $text-light-color;
    }

    &:-ms-input-placeholder {
      color: $text-light-color;
    }

    &::-webkit-input-placeholder {
      color: $text-light-color;
    }
  }
}
