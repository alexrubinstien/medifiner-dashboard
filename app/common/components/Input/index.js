import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './styles.scss';

const Input = ({
  children,
  className,
  inputClassName,
  label,
  wide,
  ...props,
}) => (
  <div
    className={
      classNames(className, {
        input: true,
        'input--wide': wide,
      })
    }
  >
    <label className="input__label">{label}</label>
    <input
      type="text"
      className={classNames('input__input', inputClassName)}
      {...props}
    />
    {children}
  </div>
);

Input.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  inputClassName: PropTypes.string,
  label: PropTypes.string,
  wide: PropTypes.bool,
};

Input.defaultProps = {
  children: null,
  className: '',
  inputClassName: '',
  label: null,
  wide: false,
};

export default Input;
