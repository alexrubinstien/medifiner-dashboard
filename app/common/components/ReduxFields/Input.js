import React from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';

import Input from '@common/components/Input';

import './styles.scss';

const FormInput = ({
  input,
  label,
  meta,
  ...props
}) => (
  <div className="field">
    <Input
      wide
      id={input.name}
      autoComplete="off"
      label={label}
      {...props}
      {...input}
      inputClassName={classNames('field__input', { 'field__input--error': (meta.touched && meta.error) })}
    />
    {(meta.touched && meta.error) && (
      <div className="field__error">
        {meta.error}
      </div>
    )}
  </div>
);

FormInput.propTypes = {
  input: PropTypes.shape({}).isRequired,
  label: PropTypes.string,
  meta: PropTypes.shape({
    error: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.array,
    ]),
    touched: PropTypes.bool,
  }).isRequired,
};

FormInput.defaultProps = {
  label: null,
};

export default FormInput;
