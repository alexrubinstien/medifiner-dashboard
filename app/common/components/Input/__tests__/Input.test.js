/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import Input from '../index.js';

describe('With Snapshot Testing', () => {
  it('Input component', () => {
    const component = renderer.create(
      <Input
        label="Label"
        placeholder="Enter Zip Code"
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Input component with custom class', () => {
    const component = renderer.create(
      <Input
        label="Label"
        placeholder="Enter Zip Code"
        className="test-input"
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Input component with custom input class', () => {
    const component = renderer.create(
      <Input
        label="Label"
        placeholder="Enter Zip Code"
        inputClassName="test-input-field"
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
