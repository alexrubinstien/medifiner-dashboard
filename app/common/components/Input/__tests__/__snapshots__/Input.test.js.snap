// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Input component 1`] = `
<div
  className="input"
>
  <label
    className="input__label"
  >
    Label
  </label>
  <input
    className="input__input"
    placeholder="Enter Zip Code"
    type="text"
  />
</div>
`;

exports[`With Snapshot Testing Input component with custom class 1`] = `
<div
  className="test-input input"
>
  <label
    className="input__label"
  >
    Label
  </label>
  <input
    className="input__input"
    placeholder="Enter Zip Code"
    type="text"
  />
</div>
`;

exports[`With Snapshot Testing Input component with custom input class 1`] = `
<div
  className="input"
>
  <label
    className="input__label"
  >
    Label
  </label>
  <input
    className="input__input test-input-field"
    placeholder="Enter Zip Code"
    type="text"
  />
</div>
`;
